mmdet/models/bbox_heads/convfc_bbox_head.py

```diff
@@ -168,6 +173,36 @@ def forward(self, x):
         return cls_score, bbox_pred
 
 
+def _view_flat_trailing_dims(x):
+    """
+    Flattens trailing dimensions
+
+    Equivalent to `x.view(x.shape[0], -1)`, but has special handling of the
+    case where `x.shape[0] == 0`
+
+    Args:
+        x (Tensor): input tensor
+
+    Returns:
+        Tensor: reshaped tensor
+
+    Example:
+        >>> import torch
+        >>> x = _view_flat_trailing_dims(torch.empty(3, 5, 7))
+        >>> assert tuple(x.shape) == (3, 35)
+        >>> x = _view_flat_trailing_dims(torch.empty(0, 5, 7))
+        >>> assert tuple(x.shape) == (0, 35)
+        >>> x = _view_flat_trailing_dims(torch.empty(0,))
+        >>> assert tuple(x.shape) == (0, 1)
+    """
+    if x.numel() == 0:
+        num_trailing = reduce(mul, x.shape[1:], 1)
```

Comment:
`reduce` and `mul` introduce two extrac packages: `functools` and `operator`, there should be a better way to implement the multiply of `x.shape[1:]`.